File: chrome/android/java/src/org/chromium/chrome/browser/autofill/AutofillFallbackSurfaceLauncher.java
// Copyright 2024 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.chrome.browser.autofill;

import android.content.Context;

import org.jni_zero.CalledByNative;
import org.jni_zero.JNINamespace;

import org.chromium.build.annotations.NullMarked;
import org.chromium.ui.base.WindowAndroid;

/** A helper class to open web pages from the keyboard accessory manual filling UI. */
@JNINamespace("autofill")
@NullMarked
public class AutofillFallbackSurfaceLauncher {

    @CalledByNative
    public static void openGoogleWalletPassesPage(WindowAndroid window) {
        Context context = window.getActivity().get();

        if (context == null) {
            return;
        }
        openGoogleWalletPassesPage(context);
    }

    public static void openGoogleWalletPassesPage(Context context) {
        GoogleWalletLauncher.openGoogleWallet(context, context.getPackageManager());
    }

    @CalledByNative
    public static void openGoogleWalletPrivatePassHelpCenterPageInCct(WindowAndroid window) {
        Context context = window.getActivity().get();

        if (context == null) {
            return;
        }
        GoogleWalletLauncher.openGoogleWalletPrivatePassHelpCenterPage(context);
    }

    private AutofillFallbackSurfaceLauncher() {}
}

File: chrome/browser/autofill/android/autofill_fallback_surface_launcher.cc
// Copyright 2024 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "chrome/browser/autofill/android/autofill_fallback_surface_launcher.h"

#include "base/android/jni_android.h"
#include "content/public/browser/web_contents.h"
#include "ui/android/view_android.h"
#include "ui/android/window_android.h"

// Must come after all headers that specialize FromJniType() / ToJniType().
#include "chrome/android/chrome_jni_headers/AutofillFallbackSurfaceLauncher_jni.h"

namespace autofill {

void ShowGoogleWalletPassesPage(content::WebContents& web_contents) {
  if (web_contents.GetNativeView() &&
      web_contents.GetNativeView()->GetWindowAndroid()) {
    Java_AutofillFallbackSurfaceLauncher_openGoogleWalletPassesPage(
        base::android::AttachCurrentThread(),
        web_contents.GetNativeView()->GetWindowAndroid()->GetJavaObject());
  }
}

void ShowGoogleWallePrivatePassesHelpCenterPageInCct(
    content::WebContents& web_contents) {
  if (web_contents.GetNativeView() &&
      web_contents.GetNativeView()->GetWindowAndroid()) {
    Java_AutofillFallbackSurfaceLauncher_openGoogleWalletPrivatePassHelpCenterPageInCct(
        base::android::AttachCurrentThread(),
        web_contents.GetNativeView()->GetWindowAndroid()->GetJavaObject());
  }
}

}  // namespace autofill

DEFINE_JNI(AutofillFallbackSurfaceLauncher)
